### PR TITLE
Additional logging for GOAWAY

### PR DESCRIPTION
**Pull request by dagnir**
## Description
Should help during debugging and troubleshooting.

## Motivation and Context
Should help during debugging and troubleshooting.

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)

## Checklist

- [x] I have read the **CONTRIBUTING** document
- [x] Local run of `mvn install` succeeds
- [x] My code follows the code style of this project
- [ ] My change requires a change to the Javadoc documentation
- [ ] I have updated the Javadoc documentation accordingly
- [x] I have read the **README** document
- [ ] I have added tests to cover my changes
- [x] All new and existing tests passed
- [ ] A short description of the change has been added to the **CHANGELOG**
- [ ] My change is to implement 1.11 parity feature and I have updated [LaunchChangelog](https://github.com/aws/aws-sdk-java-v2/blob/master/docs/LaunchChangelog.md)

## License

- [x] I confirm that this pull request can be released under the Apache 2 license

### CODE REVIEW

File: http-clients/netty-nio-client/src/main/java/software/amazon/awssdk/http/nio/netty/internal/http2/Http2MultiplexedChannelPool.java [hunk 1]
```diff
@@ -316,7 +316,7 @@ private boolean acquireStreamOnInitializedConnection(MultiplexedChannelRecord ch
     }
 
     public void handleGoAway(Channel parentChannel, int lastStreamId, GoAwayException exception) {
-        log.debug(() -> "Received GOAWAY on " + parentChannel + " with lastStreamId of " + lastStreamId);
+        log.warn(() -> "Received GOAWAY on " + parentChannel + " with lastStreamId of " + lastStreamId);
```
Comment (zoewangg) [inline http-clients/netty-nio-client/src/main/java/software/amazon/awssdk/http/nio/netty/internal/http2/Http2MultiplexedChannelPool.java]: Still not sure if we should set it to `warn` as `GOAWAY` is not uncommon and retry normally should work. Using `warn` might raise unnecessary concerns. 

Can we add the message in the exception instead?

Comment (dagnir) [reply http-clients/netty-nio-client/src/main/java/software/amazon/awssdk/http/nio/netty/internal/http2/Http2MultiplexedChannelPool.java]: Yeah looking at this again, I don't think it's necessary to increase the level here. The changes `MultiplexedChannelRecord` should be all we need

File: http-clients/netty-nio-client/src/main/java/software/amazon/awssdk/http/nio/netty/internal/http2/MultiplexedChannelRecord.java [hunk 2]
```diff
@@ -81,7 +82,15 @@ boolean acquireStream(Promise<Channel> promise) {
...
-                String message = "Connection received GOAWAY or was closed while acquiring new stream.";
+                String message;
+                // GOAWAY
+                if (state == RecordState.CLOSED_TO_NEW) {
+                    message = String.format("Connection %s received GOAWAY with Last Stream ID %d. Unable to open new "
+                                            + "streams on this connection.", connection, lastStreamId);
+                } else {
+                    message = String.format("Connection %s was closed while acquiring new stream.", connection);
+                }
+                log.warn(() -> message);
```
Comment (zoewangg) [inline http-clients/netty-nio-client/src/main/java/software/amazon/awssdk/http/nio/netty/internal/http2/MultiplexedChannelRecord.java]: Same here. Can we add the message in the exception instead?

Comment (dagnir) [reply http-clients/netty-nio-client/src/main/java/software/amazon/awssdk/http/nio/netty/internal/http2/MultiplexedChannelRecord.java]: I would rather keep this here. This is the specific case we care about right, trying to create a stream on one that has received a `GOAWAY`. The message is added to the exception as well

**Pull request merged**